source/AsyncRgbLedAnalyzerSettings.h

```diff
@@ -10,18 +10,33 @@ class AsyncRgbLedAnalyzerSettings : public AnalyzerSettings
 	AsyncRgbLedAnalyzerSettings();
 	virtual ~AsyncRgbLedAnalyzerSettings();
 
-	virtual bool SetSettingsFromInterfaces();
+	bool SetSettingsFromInterfaces() override;
 	void UpdateInterfacesFromSettings();
-	virtual void LoadSettings( const char* settings );
-	virtual const char* SaveSettings();
+	void LoadSettings( const char* settings ) override;
+	const char* SaveSettings() override;
 
-	
-	Channel mInputChannel;
-	U32 mBitRate;
+	enum Controller {
+		LED_WS2811,
+		LED_WS2812B,
+		LED_WS2813,
+		LED_TM1809,
+		LED_TM1804,
+		LED_UCS1903,
+		LED_LPD1886_8bit,
+		LED_LPD1886_12bit
+	};
 
+	Controller mLEDController = LED_WS2811;
+	Channel mInputChannel = UNDEFINED_CHANNEL;
+
+	/// bits ber LED channel, either 8 or 10 at present
+	U8 BitSize() const; 
+
+	/// LED channel count, either 3 (RGB) or 9 (three RGB outputs) at present
+	U8 LEDChannelCount() const;
 protected:
 	std::auto_ptr< AnalyzerSettingInterfaceChannel >	mInputChannelInterface;
-	std::auto_ptr< AnalyzerSettingInterfaceInteger >	mBitRateInterface;
+	std::auto_ptr< AnalyzerSettingInterfaceNumberList >	mControllerInterface;
```

Comment:
Lets change this to std::unique_ptr.
I'd like to stop using std::auto_ptr. I think we put that in there very early on (over 8 years ago) when we were removing the boost dependency from the analyzers and wanted a replacement for boost::scoped_ptr.
auto_ptr probably never made sense here. 
I'll check in on our build system support for this, and I'll have more notes in my overall review notes.

Reply:
Done, I use unique_ptr by default anyway, this was copy-paste that I missed.